### ISSUE TITLE
Too much fuel

### ISSUE THREAD

**Issue by mugling** [CleverRaven/Cataclysm-DDA#15362]
Fuel is by far too plentiful a resource in the cataclysm and/or vehicle engines are too efficient. It's perfectly possible, within a few days, to acquire a tankful of fuel and never come close to running out.

Siphoning the tanks of nearly empty abandoned vehicles into your jerrypack and looting fuel stations under the cover of darkness is both thematic and good gameplay. Therefore we perhaps need to deal with both the efficiency issue and make fuel harder to store.
#### Suggestions:
- **Reduce overall engine efficiency**
  Possibly with a few rare or otherwise balanced high efficiency variants
- **Increase effect of weight on fuel consumption**
  Encourages more variation in both vehicle parts and overall design
- **Increase effect of speed on fuel consumption**
  Also make speed more realistic overall (possibly by just rescaling the units)
- **Reduce most vehicle tank sizes**
  Balances fuel stations by limiting what you can reasonably store in a vehicle
- **Introduce more (incompatible) fuel types (LPG, propane etc)**
  Dilutes the overall supply. Also a basis for a sane implementation of engine mods

**Comment by Coolthulhu** [comment]
> Increase effect of weight on fuel consumption

I'm going this route right now. When I finish the traction system, heavy vehicles (and regular cars going offroad) will consume more fuel.

Also more suggestions:
* **Decrease the amount of available fuel**  
Gasoline is good stuff - you can use it to fuel cars, make pipe bombs, fuel flamethrowers, kickstart fires, provide light, and in a pinch cook on it. A great deal of it would be looted.
* **Have fuel rot**  
IRL gasoline doesn't last forever. After a while it starts going bad, loses some of its volatile components and becomes ill suited to use with modern optimized engines.

**Comment by mugling** [comment]
> Have fuel rot ... and become ill suited to use with modern optimized engines.

Makes sense. IRL older (and more inefficient engines) are much happier to run on poor quality fuel. Modelling would help to make vehicle design more of a trade-off rather than choosing the biggest engine you can find

**Comment by illi-kun** [comment]
> Have fuel rot

Oh God, no, just big fat __NO__.

Loosing something you gather previously due to rotting is a bad game mechanics, IMO (food is acceptable because player it's able to convert at least part of it to non-rotting versions, and I don't want to downgrade players' vehicles to coal-driven mechanisms just because coal can be stored till the end of time).

It's much better to search something new than gather the same things again and again (compare: discover new locations for some brand new things you never had before _or_ get another tank of gas because previous one was rot).

**Comment by Zireael07** [comment]
Moreover, we can't make fuel rot without making it a comestible.

**Comment by John-Candlebury** [comment]
I think that the best way to limit fuel would be the npc factions. Most factions would hoard the stuff like crazy. Especially the hell's riders and the Old Guard. If the player wants fuel to safely run a vehicle they would either have to raid those factions or ally with them.

Would produce a much more interesting gameplay that having all your fuel rot come summer months, and become unusable gunk come the second year.Sure that is realistic, but I think it would make some plausible and fun gameplay scenarios impossible, it limits the end game much more than what is IMO worthwhile

**Comment by jcd000** [comment]
I second John Candlebury's point. 
Factions holding 80% of petrol-based fuel also gives factions a bit more purpose and meshes them better with the world...

(Not to mention that then there will also be a reason to add wood gasifiers as a fuel source - Can't buy it? Make an alternative with extra work.)

**Comment by Nymall** [comment]
This makes the game harder, sure - but what does it add? Just because you have a vehicle with massive fuel loads does not mean that vehicle is impervious by far. Develop a leak without the proper tools, and that load is going to be gone very fast. Unless you're a really hard-core player or go all out bionics, a fuel leak can be a death sentence for a vehicle, and the larger the vehicle, the harder it would be to even discover a leak at all.

Factions would be a nice gateway, and would also be very lore friendly. Hell, even if there was a new merchant who simply filled fuel for goods or cash would be easy enough.

Truth be told, my major dislike of this idea stems from the fact that I tend to hunker down in a location and try to create a long-term establishment(often returning with new characters after death). This means that fuel tends to get tapped out pretty quickly, and I play stingy after that. It feels like it inadvertently punishes characters who don't do the whole roving survivor thing.

**Comment by Coolthulhu** [comment]
> Unless you're a really hard-core player or go all out bionics, a fuel leak can be a death sentence for a vehicle, and the larger the vehicle, the harder it would be to even discover a leak at all.

Fuel leaks currently aren't implemented anywhere near like that.
They're rare and obvious.

**Comment by mugling** [comment]
> It feels like it inadvertently punishes characters who don't do the whole roving survivor thing.

Answer: biodiesel

**Comment by Coolthulhu** [comment]
> Moreover, we can't make fuel rot without making it a comestible.

I was thinking less like fuel rot and more like half-life. After x time, y% fuel is removed.

**Comment by John-Candlebury** [comment]
I would save the disappearing fuel thing as a last resort nerf though.

I think that it would be best to start with:
-Reduced efficiency
-Non overt fuel leaks
-Cars needing some sort of regular upkeep of the engine and fuel storage systems.

**Comment by Coolthulhu** [comment]
> -Non overt fuel leaks

I'm against that. Having hidden issues with vehicles that you need "tricks" and careful out-of-game observation to discover sounds like a giant design problem and annoyance.

> -Cars needing some sort of regular upkeep of the engine and fuel storage systems. 

You mean "rotting" engines?
That would require a big redesign of car repairs, otherwise it would just result in tedious repairs at no cost once in a while.
Or just a "tax" for having an engine installed?

**Comment by John-Candlebury** [comment]
I think it was @kevingranade who originally made the suggestion. It involved the player getting notices in the form of your engine doing ugly noises, backfiring and not wanting to start properly. To fix it you were supposed to take a "carefully examine the car" action after which it would become clear what small part needed to be replaced/fixed. 

The non overt fuel leak would be similar, you would only notice it if you left the car parked somewhere with a non permeable floor. Then you could do the "careful examination" again and find the leak and duct tape it.

**Comment by mutt3x** [comment]
**So im the only one that likes the rotting gas idea.**
Gas would lose quality over time makes the game harder (i like) and forces you to scavenge and find other ways (more actions). You could eventualy upgrade to electric or those futuristic ones, or just start crafting that diesel. I understand some players like easy gameplay... whatever gives you the funs, nothing to add there. Maybe this can be an ON/OFF option...
**If not this than something else. Gas should have some rare and valuable aura, because of the cataclysm and all that.**

>It feels like it inadvertently punishes characters who don't do the whole roving survivor thing.

hmm, if im correct cdda is a survival game. The game is pretty much easy as is, find an RV > safely proceed to end-game. lol true.

John-Candlebury's idea is good but i dont like it in its pure form. Random merchants are good too but hey... who would buy from them (reloads shotgun)

**Comment by mugling** [comment]
> Have fuel rot
> > Oh God, no, just big fat NO.

I suppose @illi-kun is correct in that taking away already scavenged supplies is the wrong approach

> It feels like it inadvertently punishes characters who don't do the whole roving survivor thing.
> > hmm, if im correct cdda is a survival game

Indeed. Currently vehicles are so OP that it's game breaking once you know how they work and keeping it that way so you can avoid having to survive adds nothing to gameplay.

> Maybe this can be an ON/OFF option.

Generally a sign of poor design. We should aim to provide a balanced implementation that satisfies most people.

> You mean "rotting" engines?

I'd actually like to implement engine wear, with older/damaged engines leaking oil and needing topping up, belts and spark plugs requiring replacement etc.

**Comment by DeadLeaves** [comment]
A start would be to reduce the fuel existing in pumps and vehicles. Like previously said, such things are hoarded like crazy. Make finding some fuel in a vehicle something to get excited about instead of always expecting to get 4-5% fuel from each vehicle.

Another thing, while perhaps not very realistic would be to somehow limit the fuel storage capabilities of vehicles. stacking fuel tanks on your vehicle is way too easy, you can quickly get 3-4 tanks installed and then you never have to worry ever again, you're even safe from leakage since one tank only means a minor fuel loss and not the whole car useless.

**Comment by illi-kun** [comment]
@DeadLeaves I don't think it's a good idea to limitate number of tanks because player always can store fuel in containers, so your proposition will cause only inconvinience for player by adding exccesive (and annoying) loops of refueling.

**Comment by illi-kun** [comment]
So, the best way of fixing this issue is to increase fuel consumption by big engines, so you will need to decide which option is better for you: have monstrous power under the hood or save fuel and enjoy by 10 mph ride; adding more ways for spending of fuel will also help.

**Comment by kevingranade** [comment]
The rate of consumption is probably low, especially for megavehicles.

**Comment by mouseturd** [comment]
In regards to fuel rot- Don't forget about fuel stabilizer sold at most hardware stores and gas stations. This could also  be craftable (maybe a recipe in the biodiesal book)

**Comment by Malkeus** [comment]
http://arstechnica.com/civis/viewtopic.php?t=1165310
Interesting read about fuel stabilizers and 'fuel-rot' in general. Apparently gas in an air-tight container will keep for years. Cars with mostly empty fuel tanks don't qualify though. I hadn't realized how much moisture had to do with bad fuel.

**Comment by kevingranade** [comment]
The whole thing is rather tricky, the general process of "rot" is going to proceed in parallel across all petroleum products in the game, but potentially proceeding faster in some circumstances (uncovered containers, half-full fuel tanks, in-engine as opposed to in a sealed container).  The general rule of thumb seems to be that it's possible to use after a year of storage, but at that point it starts requiring admixture with "fresh" fuel to get it to work, and supposedly in increasing quantities as time goes on.

tl;dr for the proposal, as time passes, the survivor will need to start crafting some fuel around the 1.5 year mark, craft half of the fuel they use by the 2 year mark, and 75% of the fuel they use by the 4 year mark, at which point the ratio stays at 75%.

Proposal:
I'd like to institute rot based on age*, ignoring the effects of storage (as in make all fuels act as if they are properly stored).  When mixing fuels it would average the age, biased by relative volumes of course.  The effect of rot would be that when age passes one year** there is an accumulating chance of poor starting behavior and engine events (backfires for extra noise, low fuel efficiency, further TBD engine conditions), culminating in inability to start an engine at some point, and inability to keep running at some later point.  One point I'd like to add is to cap the effective age of the fuel, so 10-year-old fuel is going to act the same as fuel at the age cap, which is especially important when mixing with newer fuel, or in other words using old fuel as a extender to fresh fuel.

Let's assume for the sake of discussion that the point at which fuel can't sustain combustion is 2 years and the age cap for fuel is 4 years, but I have no idea what the real numbers should be so it's definitely up for discussion.

Effects:
Fuel found in cars and fuel stations*** (found fuel) is going to work fine up to one year** after the cataclysm.
Past one year, found fuel is going to start getting unreliable, if a player hasn't crafted any fuel, this is going to apply equally to all fuel encountered.
In order to avoid engine problems altogether, the survivor will need to mix enough crafted fuel with found fuel to lower the average age to less than one year, the ratio of crafted fuel to found fuel needed to do this is going to continuously increase.
If the fuel in a vehicle starts getting critically old, habits such as leaving the engine idling**** when a quick getaway is required will be beneficial.
Two years into the cataclysm, all found fuel is going to be unusable unless mixed with crafted fuel, and found fuel will need to be mixed with new fuel (not just crafted, but recently crafted) in a 50:50 ratio in order to prevent engine problems.
Four years into the cataclysm, all found fuel reaches a steady state of maximum degeneration, it requires mixing with new fuel in a 50:50 ratio to even have a chance of starting, and even then it's nearly 0.  Mixing found fuel with new fuel in a 25:75 ratio results in problem-free engine operation, but of course also requires producing large quantities of your own fuel.  At this point found fuel is mostly useful as either a fuel extender for fuel you craft, or possibly as a component of other crafting recipes.  On the plus side, it enters this steady state and doesn't get any worse.

Additional notes:
On the balance side, I think this is more generous than we'd like.  It isn't even going to kick in for most games, and by the time it does, the chance that you can't manage the transition with ease is negligible.
It's probably going to come as a surprise to people who build a megavehicle and then drive it around looting from cars and gas stations, we need a way for them to find out what's happening.
Fresh fuel should also be available from NPC settlements long before the bad effects kick in.
The player will need to be able to monitor fuel condition, characterizing it at least as (names up for discussion) "fresh" (less than a year old), "stale" (1-2 years old), "old" (2-4 years old) and "denatured" (4+ years old), some intermediate values would be good in order to allow for safety margins, say at 10 months or so. It would also be nice to allow the player to "top off with found fuel until tank is at desired age level".

* This wouldn't be an active process like food rot, because it's never going to turn into a different item type, and is part of the reason I'd like to ignore storage and environmental conditions. You'd just check the effective age whenever the fuel is used (or examined) and act accordingly.
** Just as a reminder, all the timespans I'm referencing here would *not* be subject to adjustment to the "season length" option, they have nothing to do with each other.
*** As a very rough adjustment for storage if we don't want to ignore it entirely, we could have fuel stored in cars at the start of the cataclysm count as slightly older, perhaps by a month or so, than fuel that is "properly stored".  This would only apply to fuel generated with mapgen, crafted fuels would get an item timestamp when crafted as usual.
**** We might want to differentiate between idling with the player driving and idling with the player not in the driver's seat at some point, managing cold-start in winter conditions in New England can be a significant issue.

**Comment by Rivet-the-Zombie** [comment]
If we're going to add fuel degradation, then we really ought to add chemical fuel stabilizers as well, so the player can store treated fuel for longer than the untreated stuff.

**Comment by kevingranade** [comment]
Part of the proposal was:
> ignoring the effects of storage (as in make all fuels act as if they are properly stored).

What I was reading indicated that the 1 year limit before engine problems *is* based on treated or properly stored fuel, and untreated and improperly stored fuels would go bad (for different reasons) within just a few months.  Is that not the case?

**Comment by John-Candlebury** [comment]
>What I was reading indicated that the 1 year limit before engine problems is based on treated or properly stored fuel, and untreated and improperly stored fuels would go bad (for different reasons) within just a few months. Is that not the case?

Yes it is.

Also is it possible to ignore geography and add possible petroleum wells/refineries to the game?

**Comment by NickGrease** [comment]
Would there likely be a point where the player simply ignores found fuel because they can craft enough on there own? Would this happen in less then a year?

**Comment by Rivet-the-Zombie** [comment]
Most chemical stabilizers can give you an extra year (or maybe two) on the fuel's lifespan before it starts to turn into varnish, particularly when it's already in the fuel tank/system.

**Comment by illi-kun** [comment]
Continued in #19348.
